admin/pre_reg/serializers.py

```diff
@@ -66,12 +66,14 @@ def get_url(draft):
 
 
 def get_embargo(draft, json_safe):
-    registration_choice = draft.approval.meta.get('registration_choice', None)
-    if registration_choice == EMBARGO:
-        time = parser.parse(draft.approval.meta['embargo_end_date'])
-        return iso8601format(time) if json_safe else time
-    return IMMEDIATE
-
+    if draft.approval is not None:
+        registration_choice = draft.approval.meta.get('registration_choice', None)
+        if registration_choice == EMBARGO:
+            time = parser.parse(draft.approval.meta['embargo_end_date'])
+            return iso8601format(time) if json_safe else time
+        return IMMEDIATE
+    else:
+        return None
```

Comment:
Does `None` serialize fine? Otherwise you might be explicit and put something like "Canceled Registration" to keep the `type` the same